packages/cache/src/lib/CacheMiddleware.ts

```diff
@@ -0,0 +1,727 @@
+
+    payload.errors = [...payload.errors, ...errors];
+
+    return payload;
+  }
+
+  public async [Method.Remove]<Value = StoredValue>(payload: Payloads.Remove.ByHook<Value>): Promise<Payloads.Remove.ByHook<Value>>;
+  public async [Method.Remove](payload: Payloads.Remove.ByValue): Promise<Payloads.Remove.ByValue>;
+
+  @PostProvider()
+  public async [Method.Remove]<Value = StoredValue>(payload: Payloads.Remove<Value>): Promise<Payloads.Remove<Value>> {
+    const { key, path, type, value } = payload;
+    const { provider: cache } = this.context;
+    const getPayload = await cache[Method.Get]({ method: Method.Get, key, path: [], errors: [] });
+
+    if (isPayloadWithData<CacheMiddleware.Document<StoredValue>>(getPayload) && (await this.checkNotExpired(getPayload.data, key))) {
+      if (isRemoveByHookPayload(payload)) {
+        const { hook } = payload;
+        const { errors } = await cache[Method.Remove]({ method: Method.Remove, key, path: ['value', ...path], type, hook, errors: [] });
+
+        payload.errors = [...payload.errors, ...errors];
+      }
+
+      if (isRemoveByValuePayload(payload)) {
+        const { errors } = await cache[Method.Remove]({ method: Method.Remove, key, path: ['value', ...path], type, value, errors: [] });
+
+        payload.errors = [...payload.errors, ...errors];
+      }
+    }
+
+    return payload;
+  }
+
+  @PostProvider()
+  public async [Method.Update]<Value = StoredValue>(payload: Payloads.Update<StoredValue, Value>): Promise<Payloads.Update<StoredValue, Value>> {
+    const { key, hook } = payload;
+    const { provider: cache } = this.context;
+    const getPayload = await cache[Method.Get]({ method: Method.Get, key, path: [], errors: [] });
+
+    payload.errors = [...payload.errors, ...getPayload.errors];
+
+    if (isPayloadWithData<CacheMiddleware.Document<StoredValue>>(getPayload) && (await this.checkNotExpired(getPayload.data, key))) {
+      const newValue = await hook(getPayload.data.value, key);
+      const { errors } = await this[Method.Set]({ method: Method.Set, key, value: newValue, path: [], errors: [] });
+
+      payload.errors = [...payload.errors, ...errors];
+    }
+
+    return payload;
+  }
+
+  private async checkNotExpired(data: CacheMiddleware.Document<unknown>, key: string) {
+    if (!this.context.ttl || !this.context.ttl.enabled) return true;
+    if (new Date().getTime() - new Date(data.created).getTime() <= (this.context.ttl.timeout ?? 1000)) return true;
+
+    const { provider: cache } = this.context;
+
+    await cache[Method.Delete]({ method: Method.Delete, key, path: [], errors: [] });
+
+    return false;
+  }
+
+  private async populateCache() {
+    const { provider: cache } = this.context;
+
+    await cache[Method.Clear]({ method: Method.Clear, errors: [] });
+
+    const { data } = await this.provider[Method.Entries]({ method: Method.Entries, data: {}, errors: [] });
+
+    if (data) {
```

Comment:
This will always resolve to `true` as you are passing `data: {}` in the payload.